src/engine/enginemaster.cpp

```diff
@@ -443,22 +546,79 @@ void EngineMaster::process(const int iBufferSize) {
     }
 
     if (masterEnabled) {
-        // Mix the three channels together. We already mixed the busses together
-        // with the channel gains and overall master gain.
-        if (!m_pMasterTalkoverMix->toBool()) {
-            // Add Talkover to Master output
-            SampleUtil::copy4WithGain(m_pMaster,
-                    m_pOutputBusBuffers[EngineChannel::LEFT], 1.0,
-                    m_pOutputBusBuffers[EngineChannel::CENTER], 1.0,
-                    m_pOutputBusBuffers[EngineChannel::RIGHT], 1.0,
+        // Make master and booth mixes from the
+        // crossfader orientation buffers and talkover mix
+        SampleUtil::copy3WithGain(m_pMaster,
+            m_pOutputBusBuffers[EngineChannel::LEFT], 1.0,
+            m_pOutputBusBuffers[EngineChannel::CENTER], 1.0,
+            m_pOutputBusBuffers[EngineChannel::RIGHT], 1.0,
+            iBufferSize);
+
+        if (configuredTalkoverMixMode != TalkoverMixMode::DIRECT_MONITOR
+            && m_pNumMicsConfigured->get() > 0) {
+              m_pInputLatencyCompensationDelay->process(m_pMaster, iBufferSize);
+        }
+
+        if (boothEnabled) {
+            CSAMPLE boothGain = m_pBoothGain->get();
+
+            if (configuredTalkoverMixMode == TalkoverMixMode::MASTER) {
+                // Mix the talkover mix with the master output, but exclude it from
+                // the booth output.
+                if (m_bRampingGain) {
+                    SampleUtil::copy1WithRampingGain(m_pBooth, m_pMaster,
```

Comment:
It seems to be more efficient to compose the master mixxx in the booth buffer. 
In this case we have an in place gain change here.